source/text/WrappedText.cpp

```diff
@@ -313,18 +323,23 @@ void WrappedText::AdjustLine(size_t &lineBegin, int &lineWidth, bool isEnd)
 	int wordCount = static_cast<int>(words.size() - lineBegin);
 	int extraSpace = wrapWidth - lineWidth;
 
+	if(lineWidth > longestWidth)
+		longestWidth = lineWidth;
+
 	// Figure out how much space is left over. Depending on the alignment, we
 	// will add that space to the left, to the right, to both sides, or to the
 	// space in between the words. Exception: the last line of a "justified"
 	// paragraph is left aligned, not justified.
 	if(alignment == Alignment::JUSTIFIED && !isEnd && wordCount > 1)
 	{
+		longestWidth = wrapWidth;
 		for(int i = 0; i < wordCount; ++i)
 			words[lineBegin + i].x += extraSpace * i / (wordCount - 1);
 	}
 	else if(alignment == Alignment::CENTER || alignment == Alignment::RIGHT)
 	{
 		int shift = (alignment == Alignment::CENTER) ? extraSpace / 2 : extraSpace;
+		longestWidth += shift;
```

Comment:
This seems like it should be `longestWidth = lineWidth + shift;`, or else it would just increase the longest width on every line. Right?

Reply:
You are right, this is incorrect. 

Now that I look at it, I don't think `longestWidth` should reflect the aligned width of the text, but rather the width of the normal unformatted text, since it is used as an aid to determine what an appropriately sized window should be. I've updated the code to reflect this. 

If we need information about the size of the rendered and formatted text, a full Rect would probably be better, but I think that's beyond the scope of this commit.